contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -377,19 +377,37 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         emit MessageExecuted(originChainID, message.messageID);
         delete receivedFailedMessageHashes[originChainID][message.messageID];
 
+        // Re-encode the payload by ABI encoding a call to the {receiveTeleporterMessage} function
+        // defined by the {ITeleporterReceiver} interface.
+        bytes memory payload = abi.encodeCall(
+            ITeleporterReceiver.receiveTeleporterMessage,
+            (originChainID, message.senderAddress, message.message)
+        );
+
         // Reattempt the message execution with all of the gas left available for execution of this transaction.
         // Use all of the gas left because this message has already been successfully delivered, and it is the
         // responsibility of the caller to provide as much gas is needed. Compared to the initial delivery, where
         // the relayer should still receive their reward even if the message execution takes more gas than expected.
         // Require that the call be successful such that in the failure case this transaction reverts and the
         // message can be retried again if desired.
-        // solhint-disable-next-line avoid-low-level-calls
-        (bool success, ) = message.destinationAddress.call(
-            abi.encodeCall(
-                ITeleporterReceiver.receiveTeleporterMessage,
-                (originChainID, message.senderAddress, message.message)
+        //
+        // Use assembly to make the low-level call to avoid unnecessary memory expansion of the return data
+        // due to an issue with the version of Solidity used. This prevents possible "return bomb" vectors
+        // where the external contract could for the caller to use an arbitrary amount of gas.
+        address target = message.destinationAddress;
```

Comment:
nit: don't think we need additional variable here and below

Reply:
They are actually needed to make them easily available within the `assembly` block. I'll add a comment explaining this.

Reply:
I'm not sure I understand this either. Could we just refer to `message.destinationAddress` directly in the below `call()`?

Reply:
Nope, the relevant documentation about variable access in assembly is here: https://docs.soliditylang.org/en/latest/assembly.html#access-to-external-variables-functions-and-libraries

The TL;DR is:
```
Local variables of value type are directly usable in inline assembly. They can both be read and assigned to.

Local variables that refer to memory evaluate to the address of the variable in memory, not the value itself.
```
We could access that memory directly using other assembly instructions, but it ends up being much easier to make those values available by assigning them to local value types prior to the assembly.

Reply:
Makes sense, thanks for the docs link